src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceService.cs

```diff
@@ -0,0 +1,294 @@
+            {
+                var namespaceDeclaration = namespaceDecls.Single();
+                Debug.Assert((object)namespaceDeclaration == (object)compilationUnit.Members.Single());
+
+                var shouldTrigger = namespaceDeclaration.Name.Span.IntersectsWith(position) 
+                    && namespaceDeclaration.Name.GetDiagnostics().All(diag => diag.DefaultSeverity != DiagnosticSeverity.Error);
+
+                shouldTrigger = shouldTrigger && 
+                    !(await ContainsPartialTypeWithMultipleDeclarationsAsync(document, namespaceDeclaration, cancellationToken)
+                    .ConfigureAwait(false));
+
+                return (shouldTrigger, shouldTrigger ? namespaceDeclaration : null);
+            }
+
+            if (namespaceDecls.Length == 0)
+            {
+                var firstMemberDeclaration = compilationUnit.Members.FirstOrDefault();
+
+                var shouldTrigger = firstMemberDeclaration != null
+                    && firstMemberDeclaration.GetNameToken().Span.IntersectsWith(position);
+
+                shouldTrigger = shouldTrigger &&
+                    !(await ContainsPartialTypeWithMultipleDeclarationsAsync(document, compilationUnit, cancellationToken)
+                    .ConfigureAwait(false));
+
+                return (shouldTrigger, null);
+            }
+
+            return default;
+        }
+
+        /// <summary>
+        /// Try to change the namespace declaration based on the refacoring rules:
+        /// 
+        ///     - if neither declared and target namespace are "" (i.e. global namespace),
+        ///     then we try to change the name of the namespace.
+        ///     - if declared namespace is "", then we try to move all types declared 
+        ///     in global namespace in the document into a new namespace declaration.
+        ///     - if target namespace is "", then we try to move all members in declared 
+        ///     namespace to global namespace (i.e. remove the namespace declaration).    
+        /// </summary>
+        protected override SyntaxNode ChangeNamespaceDeclaration(
+            SyntaxNode root, 
+            ImmutableArray<string> declaredNamespaceParts, 
+            ImmutableArray<string> targetNamespaceParts)
+        {
+            if (declaredNamespaceParts.IsDefault || targetNamespaceParts.IsDefault || !(root is CompilationUnitSyntax compilationUnit))
+            {
+                return root;
+            }
+
+            // Move everything from global namespace to a namespace declaration
+            if (declaredNamespaceParts.Length == 1 && declaredNamespaceParts[0].Length == 0)
+            {
+                var targetNamespaceDecl = SyntaxFactory.NamespaceDeclaration(
+                    name: CreateNameSyntax(targetNamespaceParts, aliasQualifier: null, targetNamespaceParts.Length - 1)
+                    .WithAdditionalAnnotations(WarningAnnotation),
```

Comment:
indentation is off.  sub-exprs shoudl be nested at least one more from their parent (in this case, an argumentsyntax).